### PR TITLE
TorchVision cocoapods package

### PR DESCRIPTION
**Pull request by elb3k**
Created Cocoapods package with `libtorchvision_ops.a` for iOS development.
Cocoapods is created with similar fashion to [LibTorch](https://github.com/pytorch/pytorch/blob/master/ios/LibTorch.podspec).

Library is compiled using torch downloaded from `LibTorch`. (Provided `build_podspec.sh` to create archive)
Currently uploaded test version of package to [Cocoapods trunk](cocoapods.org) with name `LibTorch_vision_ops`.
Need to modify author and homepage on `LibTorch_vision.ops.podspec` file to main branch. (Currently on fork).

Once PR Is accepted, I will pull package from Cocoapods.

### CODE REVIEW

**Comment by fmassa** [comment]
@husthyc can you have a look?

**Comment by husthyc** [comment]
@elb3k Thanks for drafting this PR. We are planning to do the Cocoapod release in recent days which aligns with the main PyTorch 1.9.0 release day. 

I think we cannot build the lib on a local machine and push to Cocoapod because if we do that, the debug symbols in the static lib will contain your personal machine's paths. Basically what we will do is we will trigger a job on the CI machine and upload the zip to AWS s3 bucket from there.

Can you hold this PR for some days and see if our official release works? Thanks.

**Comment by elb3k** [comment]
> @elb3k Thanks for drafting this PR. We are planning to do the Cocoapod release in recent days which aligns with the main PyTorch 1.9.0 release day.
> 
> I think we cannot build the lib on a local machine and push to Cocoapod because if we do that, the debug symbols in the static lib will contain your personal machine's paths. Basically what we will do is we will trigger a job on the CI machine and upload the zip to AWS s3 bucket from there.
> 
> Can you hold this PR for some days and see if our official release works? Thanks.

@husthyc Yes.

File: .gitignore [hunk 1]
```diff
@@ -36,3 +36,5 @@ gen.yml
 
 ## Xcode User settings
 xcuserdata/
+
+.DS_Store
```
Comment (datumbox) [inline .gitignore:40]: nit: Add new line at the end of file.

File: ios/.gitignore [hunk 2]
```diff
@@ -0,0 +1,4 @@
+torch
+lib
+LICENSE
+libtorchvision_ios.zip
```
Comment (datumbox) [inline ios/.gitignore:4]: nit: Add new line at the end of file.

File: ios/LibTorch_vision_ops.podspec [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+  s.description      = <<-DESC
+Additional TorchVision operations for LibTorch
+                       DESC
+
+  s.homepage         = 'https://github.com/elb3k/vision'
+  # s.screenshots     = 'www.example.com/screenshots_1', 'www.example.com/screenshots_2'
+  s.license          = { :type => 'MIT', :file => 'LICENSE' }
+  s.author           = 'elbekkhoshimjonov@gmail.com'
+  s.source           = { :http => "https://github.com/elb3k/vision/releases/download/v#{s.version}/libtorchvision_ops_ios_#{s.version}.zip" }
```
Comment (datumbox) [inline ios/LibTorch_vision_ops.podspec:28]: Update the above information to point to the main repo

**Review by datumbox — COMMENTED** [review]
Thanks for the PR. I added a few comments about nits I spotted, let me know what you think.

File: ios/LibTorch_vision_ops.podspec (hunk 3 above)
Comment (elb3k) [reply ios/LibTorch_vision_ops.podspec]: Updated information

**Comment by fmassa** [comment]
Hi,

Thanks a lot for the PR!

We have released a cocoapods package with the 0.10.0 release, and it has been merged in https://github.com/pytorch/vision/pull/4055

Thanks again for your help!

**Comment by elb3k** [comment]
> Hi,
> 
> Thanks a lot for the PR!
> 
> We have released a cocoapods package with the 0.10.0 release, and it has been merged in #4055
> 
> Thanks again for your help!

Hi, I am glad to hear that.

**Pull request closed without merging**